src/components/modals/IssueErc20TokenModal.tsx

```diff
@@ -1,23 +1,48 @@
 import { t, Trans } from '@lingui/macro'
 import { Form, Input } from 'antd'
 import { useForm } from 'antd/lib/form/Form'
+import { CV_V1, CV_V1_1, CV_V2, CV_V3 } from 'constants/cv'
+import { ProjectMetadataContext } from 'contexts/projectMetadataContext'
 import { TransactorInstance } from 'hooks/Transactor'
-import { useState } from 'react'
+import { useV1IssueErc20TokenTx } from 'hooks/v1/transactor/IssueErc20TokenTx'
+import { useV2IssueErc20TokenTx } from 'hooks/v2/transactor/V2IssueErc20TokenTx'
+import { useV3IssueErc20TokenTx } from 'hooks/v3/transactor/V3IssueErc20TokenTx'
+import { useContext, useState } from 'react'
 
 import { emitErrorNotification } from 'utils/notifications'
 import { IssueErc20TokenTxArgs } from '../IssueErc20TokenButton'
 import TransactionModal from '../TransactionModal'
 
+/**
+ * Return the appropriate issue erc20 token hook for the given contract version [cv].
+ * @returns
+ */
+const useIssueErc20TokenTx = ():
```

Comment:
A similar outcome can be achieved with Context, but figured a simple hook dependent on ProjectMetadataContext is nice enough for now.